File: Assignment_4/java/flyweight/src/TextType.java
public class TextType {

    FontType fontType;
    int fontSize;
    ColorType colorType;

    TextType(FontType font_type, int font_size, ColorType color_type) {
        fontType = font_type;
        fontSize = font_size;
        colorType = color_type;
    }

    FontType GetFontType() {
        return fontType;
    }

    int GetFontSize() {
        return fontSize;
    }

    ColorType GetColorType() {
        return colorType;
    }

    String GetPropertiesAsString() {
        String properties = fontType.toString() + " " + fontSize + " " + colorType.toString();
        return properties;
    }
}

File: Assignment_4/cpp/flyweight/TextType.cpp
//
// Created by Sean on 4/21/2023.
//

#include "TextType.h"

TextType::TextType(FontType font, int font_size, ColorType color_type)
  : font(font)
  , fontSize(font_size)
  , colorType(color_type)
{
}

const FontType & TextType::GetFont()
{
  return font;
}

const int & TextType::GetFontSize()
{
  return fontSize;
}

const ColorType & TextType::GetColorType()
{
  return colorType;
}